src/firebase_functions/scheduler_fn.py

```diff
@@ -103,10 +103,25 @@ def on_schedule_wrapped(request: _Request) -> _Response:
             if schedule_time_str is None:
                 schedule_time = _dt.datetime.utcnow()
             else:
-                schedule_time = _dt.datetime.strptime(
-                    schedule_time_str,
-                    "%Y-%m-%dT%H:%M:%S%z",
-                )
+                try:
+                    # Try to parse with the stdlib which supports fractional
+                    # seconds and offsets in Python 3.11+ via fromisoformat.
+                    # Normalize RFC3339 'Z' to '+00:00' for fromisoformat.
+                    iso_str = schedule_time_str
+                    if iso_str.endswith("Z"):
+                        iso_str = iso_str[:-1] + "+00:00"
+                    schedule_time = _dt.datetime.fromisoformat(iso_str)
+                except ValueError:
+                    # Fallback to strict parsing without fractional seconds
+                    try:
+                        schedule_time = _dt.datetime.strptime(
+                            schedule_time_str,
+                            "%Y-%m-%dT%H:%M:%S%z",
+                        )
+                    except ValueError as e:
+                        # If all parsing fails, log and use current UTC time
+                        _logging.exception(e)
+                        schedule_time = _dt.datetime.utcnow()
```

Comment:
I think this will return a naive UTC?

https://docs.python.org/3.11/library/datetime.html

but some branches of this code set schedule_time as an aware UTC. 

It might be safest to use `datetime.now(timezone.utc)` in the fallback, as in the docs above